### PR TITLE
feat(entry files): allows to redefine sets of the specs for distribut…

### PR DESCRIPTION
**Pull request by jbubsk**
…ion between browsers

- created utils.js and moved there isSpecFile function
- extracted splitArray to utils.js
- allowed to customize getSets function in sharding config
- added to README.md example for getSets customization

### CODE REVIEW

File: lib/framework.js [hunk 1]
```diff
@@ -27,23 +26,17 @@ function setupCoverageReporting(fullConfig) {
...
-function splitArray(flatArray, numCols){
-  const maxColLength = Math.ceil(flatArray.length/numCols);
-  const nestedArray = _.chunk(flatArray, maxColLength);
-  for (var i = nestedArray.length; i < numCols; i++) {
-    nestedArray.push([]);
-  }
-  return nestedArray;
+function setupSets(config, basePath, files) {
+  config.sets = config.getSets(config, basePath, files);
+  var lastSet = config.sets && config.sets.length && config.sets[config.sets.length - 1];
```
Comment (dreef3) [inline lib/framework.js]: If we use lodash, why not simplify this expression as well? `_.last(config.sets)`

File: README.md [hunk 2]
```diff
@@ -23,7 +23,10 @@ You can pass configuration to override these defaults:
 {
   sharding: {
     specMatcher: /(spec|test)s?\.js/i,
-    base: '/base'
+    base: '/base',
+    getSets: function(config, basePath, files) {
+        return files.served.map(/*any mapper*/);
```
Comment (dreef3) [inline README.md]: Would be nice to see some close to real-world mapper example too

**Review by dreef3 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jbubsk) [reply README.md]: I added more extended one

File: lib/framework.js (hunk 1 above)
Comment (jbubsk) [reply lib/framework.js]: Good idea

**Comment by rschuft** [comment]
The README file might benefit from an explanation of when you might want to override getSets.

**Comment by rschuft** [comment]
Please add yourself to the contributors key in package.json:

```javascript
"contributors": [
    {
      "name": "Ryan Schuft",
      "email": "ryan.schuft@gmail.com",
      "url": "https://github.com/rschuft"
    }
],
```

**Review by rschuft — APPROVED** [review]
This looks good, I'll merge it and bump the minor version.

**Pull request merged**